lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -518,7 +518,7 @@ def _get_meta_from_installed_dir(
     try:
         with open(b_manifest_json_path, 'rb') as manifest_fd:
             b_manifest_txt = manifest_fd.read()
-    except OSError:
+    except (IOError, OSError):
```

Comment:
Should we do some sort of a python-dependent catching?
```suggestion
    no_file_excs = OSError,
    if PY2:
        no_file_excs += IOError,

    try:
        with open(b_manifest_json_path, 'rb') as manifest_fd:
            b_manifest_txt = manifest_fd.read()
    except no_file_excs:
```